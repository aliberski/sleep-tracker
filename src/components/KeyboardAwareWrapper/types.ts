import { ReactNode } from 'react';
import { ViewStyle } from 'react-native';

export interface IProps {
  backgroundColor?: string;
  padding?: number;
  children: ReactNode;
  center?: boolean;
  extraScrollHeight?: number;
  style?: any;
}

export interface IStyle {
  contentContainer: ViewStyle;
  center: ViewStyle;
}
