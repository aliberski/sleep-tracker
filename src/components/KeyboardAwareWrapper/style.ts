import { StyleSheet } from 'react-native';
import { IStyle } from './types';

export default StyleSheet.create<IStyle>({
  contentContainer: {
    flexGrow: 1,
  },
  center: {
    justifyContent: 'center',
  },
});
