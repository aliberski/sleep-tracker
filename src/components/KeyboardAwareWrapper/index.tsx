import React from 'react';
import { StyleSheet } from 'react-native';
import { KeyboardAwareScrollView } from 'react-native-keyboard-aware-scroll-view';

import { IProps } from './types';
import style from './style';

const KeyboardAwareWrapper = (props: IProps) => {
  const {
    backgroundColor,
    padding,
    children,
    center,
    extraScrollHeight,
  } = props;

  const containerStyle = StyleSheet.flatten([
    { backgroundColor, padding },
    props.style,
  ]);
  const contentContainerStyle = StyleSheet.flatten([
    style.contentContainer,
    center && style.center,
  ]);

  return (
    <KeyboardAwareScrollView
      keyboardShouldPersistTaps='handled'
      style={containerStyle}
      contentContainerStyle={contentContainerStyle}
      extraScrollHeight={extraScrollHeight}
      scrollEnabled
      enableOnAndroid
    >
      {children}
    </KeyboardAwareScrollView>
  );
};

export default KeyboardAwareWrapper;
