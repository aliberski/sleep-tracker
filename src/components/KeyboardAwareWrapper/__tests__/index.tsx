import { Text } from 'react-native';
import React from 'react';

import KeyboardAwareWrapper from '../index';
import renderer from 'react-test-renderer';

describe('KeyboardAwareWrapper', () => {
  it('renders correctly', () => {
    const tree = renderer
      .create(
        <KeyboardAwareWrapper center>
          <Text>test</Text>
        </KeyboardAwareWrapper>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
