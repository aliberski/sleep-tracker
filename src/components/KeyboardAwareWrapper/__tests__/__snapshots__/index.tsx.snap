// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`KeyboardAwareWrapper renders correctly 1`] = `
<RCTScrollView
  automaticallyAdjustContentInsets={false}
  contentContainerStyle={
    Object {
      "flexGrow": 1,
      "justifyContent": "center",
    }
  }
  contentInset={
    Object {
      "bottom": 0,
    }
  }
  enableAutomaticScroll={true}
  enableOnAndroid={true}
  enableResetScrollToCoords={true}
  extraHeight={75}
  extraScrollHeight={0}
  getScrollResponder={[Function]}
  handleOnScroll={[Function]}
  keyboardDismissMode="interactive"
  keyboardOpeningTime={250}
  keyboardShouldPersistTaps="handled"
  keyboardSpace={0}
  onScroll={[Function]}
  resetKeyboardSpace={[Function]}
  scrollEnabled={true}
  scrollEventThrottle={1}
  scrollForExtraHeightOnAndroid={[Function]}
  scrollIntoView={[Function]}
  scrollToEnd={[Function]}
  scrollToFocusedInput={[Function]}
  scrollToPosition={[Function]}
  showsVerticalScrollIndicator={true}
  style={
    Object {
      "backgroundColor": undefined,
      "padding": undefined,
    }
  }
  update={[Function]}
  viewIsInsideTabBar={false}
>
  <View>
    <Text>
      test
    </Text>
  </View>
</RCTScrollView>
`;
